packages/babel-traverse/test/evaluation.js

```diff
@@ -37,4 +37,11 @@ suite("evaluation", function () {
       false
     );
   });
+
+  test("should not be confident when var is redeclared in the same scope", function () {
+    assert.strictEqual(
+      getPath("var x = 2; var y = x + 2; { var x = 3 }").get("body.1.declarations.0.init").evaluate().confident,
```

Comment:
It deopts when there is one more assignment in any of the branches. So I don't think it'll become `y=4`.